### PR TITLE
resourceLookup for python client

### PR DESCRIPTION
**Pull request by sutartmelson**

### CODE REVIEW

**Comment by sutartmelson** [comment]
Wasn't sure how to format lines 499 through 512 in lib_test.py, but this got the tests passing.

File: clients/python/girder_client/__init__.py [hunk 1]
```diff
@@ -316,6 +316,19 @@ def getResource(self, path, id=None, property=None):
 
         return self.get(route)
 
+    def resourceLookup(self, path, test=False):
+        """
+        Look up a and retrieve resource in the data hierarchy by path.
```
Comment (zachmullen) [inline clients/python/girder_client/__init__.py]: > Look up and retrieve a

File: tests/cases/py_client/lib_test.py [hunk 2]
```diff
@@ -467,3 +467,56 @@ def _patchRaw(url, request):
...
+        # Test invalid path, default
+        try:
+            self.client.resourceLookup(testPath)
+        except girder_client.HttpError as e:
+            self.assertEqual(e.status, 400)
+            self.assertEqual(e.method, 'GET')
+            resp = json.loads(e.responseText)
+            self.assertEqual(resp['type'], 'rest')
+            self.assertEqual(resp['message'],
+                             'Path not found: %s' % (testInvalidPath))
```
Comment (zachmullen) [inline tests/cases/py_client/lib_test.py]: We don't assert that the `except` block was actually entered. Instead, let's do:

```
with self.assertRaises(girder_client.HttpError) as cm:
    self.resourceLookup(testPath)

self.assertEqual(cm.exception.status, 400)
...
```

File: tests/cases/py_client/lib_test.py [hunk 3]
```diff
@@ -467,3 +467,56 @@ def _patchRaw(url, request):
...
+        # Test valid path, test = False
+        self.assertEqual(self.client.resourceLookup(testPath,
+                                                    test=False)['_id'],
+                         item['_id'])
+
+        # Test invalid path, test = False
+        try:
+            self.client.resourceLookup(testPath, test=False)
+        except girder_client.HttpError as e:
+            self.assertEqual(e.status, 400)
```
Comment (zachmullen) [inline tests/cases/py_client/lib_test.py]: Same thing for this block

File: tests/cases/py_client/lib_test.py [hunk 4]
```diff
@@ -485,15 +485,15 @@ def testResourceLookup(self):
...
-            self.assertEqual(e.status, 400)
-            self.assertEqual(e.method, 'GET')
-            resp = json.loads(e.responseText)
-            self.assertEqual(resp['type'], 'rest')
-            self.assertEqual(resp['message'],
-                             'Path not found: %s' % (testInvalidPath))
+
+        self.assertEqual(cm.exception.status, 400)
+        self.assertEqual(e.method, 'GET')
+        resp = json.loads(e.responseText)
```
Comment (zachmullen) [inline tests/cases/py_client/lib_test.py]: `e` should be `cm.exception`

File: tests/cases/py_client/lib_test.py [hunk 5]
```diff
@@ -467,3 +467,56 @@ def _patchRaw(url, request):
...
+        self.assertEqual(cm.method, 'GET')
+        resp = json.loads(cm.responseText)
+        self.assertEqual(resp['type'], 'rest')
+        self.assertEqual(resp['message'],
+                         'Path not found: %s' % (testInvalidPath))
+
+        # Test valid path, test = True
+        self.assertEqual(self.client.resourceLookup(testPath,
+                                                    test=True)['_id'],
+                         item['_id'])
```
Comment (zachmullen) [inline tests/cases/py_client/lib_test.py]: Easier to read as

```
self.assertEqual(
    self.client.resourceLookup(testPath, test=True)['_id'],
    item['_id'])
```

Comment (sutartmelson) [inline tests/cases/py_client/lib_test.py]: I don't think the tests pass like that though

Comment (zachmullen) [inline tests/cases/py_client/lib_test.py]: They should, by my count it's under the line length limit.

**Comment by zachmullen** [comment]
LGTM

**Pull request merged**